controllers/userController.ts

```diff
@@ -51,3 +51,38 @@ export const update =
             return next(error);
         }
     };
+
+// Delete controller
+// ----------------------------------------------------------------------------
+
+export const deleteUser =
+    (deleteUser: UserMethods.DeleteUser): RequestHandler =>
+    async (
+        req: Request,
+        res: Response,
+        next: NextFunction
+    ): Promise<void | Response> => {
+        try {
+            // Check if the request has a session
+            if (req?.session?.user?.id) {
```

Comment:
I have a question here. How to handle the situation when the admin wants to delete a user?

Reply:
I suggest writing a set of admin only routes to handle tasks which can't be done otherwise. I'd prefer to keep the extra security the current implementation offers, personally.

Reply:
Agree. A set of admin-only routes make sense.